tests/nrf5340_audio/lc3_file/src/main.c

```diff
@@ -0,0 +1,348 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/fff.h>
+#include <modules/lc3_file.h>
+#include <zephyr/fs/fs.h>
+
+DEFINE_FFF_GLOBALS;
+
+FAKE_VALUE_FUNC(int, sd_card_init);
+FAKE_VALUE_FUNC(int, sd_card_open, const char *, struct fs_file_t *);
+FAKE_VALUE_FUNC(int, sd_card_close, struct fs_file_t *);
+FAKE_VALUE_FUNC(int, sd_card_read, char *, size_t *, struct fs_file_t *);
+
+#define FRAME_BUFFER_SIZE 40
+
+static int read_lc3_fake_bytes_read;
+
+static void test_setup(void *f)
+{
+	RESET_FAKE(sd_card_init);
+	RESET_FAKE(sd_card_open);
+	RESET_FAKE(sd_card_close);
+	RESET_FAKE(sd_card_read);
+
```

Comment:
I haven't seen this before - but I do like the idea! But `DO_FOREACH_FAKE` is something that must be implemented in this file, and not available in a FFF header or something? As it would only be used here with the current structure I'm not sure if the abstraction makes it better

Reply:
Sorry yes it would be implemented in the fakes.h as follows:

/* List of fakes used by this unit tester */
#define DO_FOREACH_FAKE(FUNC)                                                                      \
	do {                                                                                       \
		FUNC(sd_card_init)                                                                \
		FUNC(sd_card_open)                                                              \
		FUNC(sd_card_close)                                                         \
		FUNC(sd_card_read)                                                         \
	} while (0)